lib/types/number.js

```diff
@@ -1,16 +1,20 @@
+"use strict";
 var random = require('../core/random');
-var MIN_INTEGER = -100000000, MAX_INTEGER = 100000000;
+var MIN_INTEGER = -100000000, MAX_INTEGER = 100000000, FLOAT_DIFFERENCE = 0.1;
 function numberType(value) {
     var min = typeof value.minimum === 'undefined' ? MIN_INTEGER : value.minimum, max = typeof value.maximum === 'undefined' ? MAX_INTEGER : value.maximum, multipleOf = value.multipleOf;
     if (multipleOf) {
         max = Math.floor(max / multipleOf) * multipleOf;
         min = Math.ceil(min / multipleOf) * multipleOf;
     }
+    // if both numbers are integers the difference should be an integer too,
+    // otherwise having values like { min: 1, max: 1.5 } will produce invalid values (NaN)
+    var fixedUnit = ((min % 1) === 0 && (max % 1) === 0) ? 1 : FLOAT_DIFFERENCE;
     if (value.exclusiveMinimum && value.minimum && min === value.minimum) {
-        min += multipleOf || 1;
+        min += multipleOf || fixedUnit;
     }
     if (value.exclusiveMaximum && value.maximum && max === value.maximum) {
-        max -= multipleOf || 1;
+        max -= multipleOf || fixedUnit;
     }
```

Comment:
Decimals are perfect for `number` values which stand for non integers, but applying the same rules exactly but for `integer` instead which value would output?